deps/rabbitmq_auth_backend_oauth2/src/uaa_jwt.erl

```diff
@@ -67,21 +84,33 @@ decode_and_verify(Token) ->
 
 -spec get_jwk(binary()) -> {ok, map()} | {error, term()}.
 get_jwk(KeyId) ->
-    case signing_keys() of
-      undefined -> {error, signing_keys_not_configured};
-      Keys      ->
-        case maps:get(KeyId, Keys, undefined) of
-            undefined ->
-                {error, key_not_found};
-            {Type, Value} ->
-                case Type of
-                    json     -> uaa_jwt_jwk:make_jwk(Value);
-                    pem      -> uaa_jwt_jwk:from_pem(Value);
-                    pem_file -> uaa_jwt_jwk:from_pem_file(Value);
-                    map      -> uaa_jwt_jwk:make_jwk(Value);
-                    _        -> {error, unknown_signing_key_type}
-                end
-        end
+    get_jwk(KeyId, true).
+
+get_jwk(KeyId, AllowUpdateJwks) ->
+    Keys = signing_keys(),
+    case maps:get(KeyId, Keys, undefined) of
+        undefined ->
+            if
+                AllowUpdateJwks ->
+                    case update_jwks_signing_keys() of
+                        ok ->
+                            get_jwk(KeyId, false);
+                        {error, no_jwks_url} ->
+                            {error, key_not_found};
```

Comment:
`no_jwks_url` indicates that the user isn't interested in JWKS, so `key_not_found` is probably a more helpful error message..

Reply:
Sounds reasonable.